src/main/java/compal/commons/Compal.java

```diff
@@ -0,0 +1,157 @@
+     */
+    public Compal() {
+        System.out.println("Compal:LOG: In Compal Constructor");
+        //Instantiate objects
+        tasklist = new TaskList(this);
+
+        storage = new StorageManager();
+
+        tasklist.arrlist = new ArrayList<>();
+        tasklist.arrlist = storage.loadCompal();
+
+        ui = new UiPart(this, tasklist.arrlist);
+
+        //start parsing commands
+        parser = new ParserManager(this, tasklist);
+    }
+    //----------------------->
+
+
+    //***MISC FUNCTIONS***----------------------------------------------------------------------------------------------
+    //------------------------------------------------------------------------------------------------------------------
+    //----------------------------------------------------------------------------------------------------------------->
+
+    //----------------------->
+
+    /**
+     * Lists all tasks that are incomplete and due in number of days input by user,
+     * or by default 7 days, as well as tasks with reminders set as true.
+     * Will print colour-coded and sorted by importance/priority.
+     */
+    public void viewReminder() {
+        int numberOfDays = 7;
+
+        ArrayList<Task> reminder = new ArrayList<>();
+        Date currentDate = Calendar.getInstance().getTime();
+
+        Calendar c = Calendar.getInstance();
+
+        c.setTime(currentDate);
+        c.add(Calendar.DATE, numberOfDays);
+        Date dateAfter = c.getTime();
+
+        c.setTime(currentDate);
+        Date dateToday = c.getTime();
+
+        for (Task t : tasklist.arrlist) {
+            Calendar tempTaskDateAndTime = Calendar.getInstance();
+            tempTaskDateAndTime.setTime(t.getDate());
+            int endingHour = Integer.parseInt(t.getStringEndTime().substring(0, 2));
+            tempTaskDateAndTime.set(Calendar.HOUR_OF_DAY, endingHour);
+            int endingMinute = Integer.parseInt(t.getStringEndTime().substring(2, 4));
+            tempTaskDateAndTime.set(Calendar.MINUTE, endingMinute);
+
+            Date deadline = tempTaskDateAndTime.getTime();
+
+
+            if (deadline != null && !t.isDone && deadline.after(dateToday)
+                    && (deadline.before(dateAfter) || t.hasReminder())) {
+                t.calculateAndSetPriorityScore();
+                reminder.add(t);
+            }
+        }
+
+        //sort/compare by task priority score
+        Comparator<Task> compareByDateTime = (Task t1, Task t2) ->
+                Long.compare(t2.getPriorityScore(), t1.getPriorityScore());
+        reminder.sort(compareByDateTime);
+
+        //clear secondary window
+        ui.clearSecondary();
+
+        //display the results
+        if (reminder.isEmpty()) {
+            ui.printg("You currently have no tasks that have reminders set or are due within "
+                            + numberOfDays + " days!",
+                    "verdana", 15, Color.DARKGREEN);
```

Comment:
Try to apply more SLAP such that you would not need comments to tell the reader what you are trying to do. The comments, as stated on the module website, is unnecessary.